include/fiction/algorithms/simulation/sidb/quickexact.hpp

```diff
@@ -0,0 +1,506 @@
+
+            //  Determine all SiDBs that have to be negatively charged to fulfill the population stability. This is an
+            //  efficient way to prune the search space by 2^k with k being the number of detected negatively-charged
+            //  SiDBs.
+
+            //  Determine if three state simulation (i.e. positively charged SiDBs can occur) is required.
+            const bool three_state_simulation_required =
+                params.base_number_detection == automatic_base_number_detection::ON &&
+                charge_lyt.three_state_simulation_required();
+
+            // If layout has at least two SiDBs, the code inside this if-statement is executed.
+            if (number_of_sidbs > 1)
+            {
+                generate_layout_without_negative_sidbs();
+            }
+
+            if (!all_sidbs_in_lyt_without_negative_detected_ones.empty() && number_of_sidbs > 1)
+            {
+                // The first cell from all_sidbs_in_lyt_without_negative_detected_ones is chosen as the dependent cell
+                // to initialize the layout (detected negatively-charged SiDBs were erased in the step before).
+                charge_distribution_surface charge_lyt_with_assigned_dependent_cell{
+                    layout, params.physical_parameters, sidb_charge_state::NEUTRAL,
+                    all_sidbs_in_lyt_without_negative_detected_ones[0]};
+
+                charge_lyt_with_assigned_dependent_cell.assign_local_external_potential(
+                    params.local_external_potential);
+                charge_lyt_with_assigned_dependent_cell.assign_global_external_potential(params.global_potential);
+
+                if constexpr (has_get_sidb_defect_v<Lyt>)
+                {
+                    for (const auto& [cell, defect] : real_placed_defects)
+                    {
+                        charge_lyt_with_assigned_dependent_cell.add_defect(cell, defect);
+                    }
+                }
+
+                // IMPORTANT: The detected negatively-charged SiDBs (they have to be negatively charged to fulfill the
+                // population stability) are considered as negatively-charged defects in the layout. Hence, there are no
+                // "real" defects assigned, but in order to set some SiDBs with a fixed negative charge, this way of
+                // implementation is chosen.
+                for (const auto& cell : detected_negative_sidbs)
+                {
+                    charge_lyt_with_assigned_dependent_cell.add_defect(
+                        cell, sidb_defect{sidb_defect_type::UNKNOWN, -1,
+                                          charge_lyt_with_assigned_dependent_cell.get_phys_params().epsilon_r,
+                                          charge_lyt_with_assigned_dependent_cell.get_phys_params().lambda_tf});
+                }
+
+                // Update all local potentials, system energy and physically validity. Flag is set to "false" to allow
```

Comment:
Which flag is set to "false"? I'm not seeing that in the code.

Reply:
That's a mistake. Thanks.